charts/loki/templates/ingress.yaml

```diff
@@ -1,11 +1,13 @@
 {{- if .Values.ingress.enabled -}}
 {{- $fullName := include "loki.fullname" . -}}
 {{- $svcPort := .Values.service.port -}}
-{{- if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
-apiVersion: networking.k8s.io/v1beta1
-{{- else -}}
-apiVersion: extensions/v1beta1
+{{- $apiVersion := "extensions/v1" -}}
+{{- if semverCompare ">=1.22-0" .Capabilities.KubeVersion.GitVersion -}}
+{{- $apiVersion = "networking.k8s.io/v1" -}}
+{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+{{- $apiVersion = "networking.k8s.io/v1beta1" -}}
```

Comment:
Could you move this under templates/_helpers.tpl ?

Reply:
I've moved in last commit, but seems we still need to assign it to variable, to work inside `if` comparisons like this:
```
            {{- if eq $apiVersion "networking.k8s.io/v1" }}
            pathType: Prefix
              {{- end }}
```
Because it doesn't want to work with template values:
```
{{- if eq (template "ingress.apiVersion" .) "networking.k8s.io/v1" }}
```
produces error `unexpected <template> in parenthesized pipeline`, and
```
{{- if eq (include "ingress.apiVersion" .) "networking.k8s.io/v1" }}
```
produces other error: `can't evaluate field Capabilities in type string`.
Maybe you can give me some workaround to this?